### PR TITLE
Add loongarch64 architecture support

### PR DESCRIPTION
**Pull request by loongson-zn**
Add loongarch64 architecture support， please review, thanks.   @yetist @xen0n @zhuyaliang @cnmushiba @zwaizwai

### CODE REVIEW

File: src/sfl.h [hunk 1]
```diff
@@ -133,7 +133,8 @@
     || defined (__s390x__) || (defined (__sparc__) && defined (__arch64__)) \
     || defined (__ia64) || defined (__itanium__) || defined (_M_IA64) \
     || defined (__riscv64) || (defined (__riscv_xlen) && __riscv_xlen == 64) \
-    || (defined (__mips__) && defined (_MIPSEL) && _MIPS_SIM == _ABI64))
+    || (defined (__mips__) && defined (_MIPSEL) && _MIPS_SIM == _ABI64) \
+    || defind (__loongarch64))
```
Comment (zhuyaliang) [inline src/sfl.h]: 拼写错误，建议下次修改后，编译一下代码

Comment (loongson-zn) [reply src/sfl.h]: 多谢，已修改

**Comment by loongson-zn** [comment]
@zhuyaliang @loongarch64/dev-team 
已重新运行，及测试结果：
```
zhangna@loongson-pc:~/lucky-github/gsl/src$ make check                               [49/1847]
./gsl testall
GSL/4.1.5 Copyright (c) 1996-2016 iMatix Corporation
              GSL Developers 2016-2017
gsl/4 I: Processing testall...
gsl/4 M: Starting regression tests...
gsl/4 M: Non-template mode:
The current script is: testall.gsl
The date and time are: 2021/07/07, 14:08:34.
This are some special characters: ! ( ) \ ? . : ;
gsl/4 M: Non-template mode:
The current script is: testall.gsl
The date and time are: 2021/07/07, 14:08:34.
This are some special characters: ! ( ) \ ? . : ;
gsl/4 M: 1: me = me
gsl/4 M: 2: blue = blue
gsl/4 M: 3: undefined variable as expected
gsl/4 M: 4: medium rare, purple = purple
gsl/4 M: Should be 'level' -> level
gsl/4 M: Should be 'child' -> child
gsl/4 M: Should be 'root'  -> root
gsl/4 M: Should be 'gsl'   -> gsl
gsl/4 M: Should be '1':    -> 1
gsl/4 M: Should be '2':    -> 2
gsl/4 M: 5: inside include file: testall.inc
gsl/4 M: 6: ?= assignment works okay
gsl/4 M: 7: var1 is undefined: okay
gsl/4 M: 8: function call with arguments seems to work
gsl/4 M: 9: function returning value
gsl/4 M: 10: Count function result (should be 2): 2
gsl/4 M: 11: last step in testall
./gsl teststr
GSL/4.1.5 Copyright (c) 1996-2016 iMatix Corporation
              GSL Developers 2016-2017
gsl/4 I: Processing teststr...
gsl/4 M: certify: should be 'Deux cent douze millions': Deux cent douze millions
gsl/4 M: certify: should be 'Two hundred and twelve million': Two hundred and twelve  million
gsl/4 M: A fool and his guilt are soon parted.
gsl/4 M: A fool and his honey are soon parted.
gsl/4 M: A fool and his guilt are soon parted.
gsl/4 M: A beautiful eye makes silence eloquent, a
kind eye makes contradiction an assent, an
enraged eye makes beauty deformed. This
little member gives life to every part about
us; and I believe the story of Argus implies
no more, than the eye is in every part; that
is to say, every other part would be
mutilated, were not its force represented
more by the eye than even by itself. --
Addison
gsl/4 M: 887
gsl/4 M: 162
gsl/4 M: 85
gsl/4 M: 1
gsl/4 M: 0
gsl/4 M: pipo
gsl/4 M: 
gsl/4 M: 85416819
gsl/4 M: tito des altes
gsl/4 M: 0
gsl/4 M: 4
gsl/4 M: 0
gsl/4 M: 1
gsl/4 M: P100
gsl/4 M: 3
  |  A beautiful eye       |  A beautiful eye makes silence   |
  |  makes silence         |  eloquent, a kind eye makes      |
  |  eloquent, a kind eye  |  contradiction an assent, an     |
  |  makes contradiction   |  enraged eye makes beauty        |
  |  an assent, an         |  deformed. This little member    |
  |  enraged eye makes     |  gives life to every part about  |
  |  beauty deformed.      |  us; and I believe the story of  |
  |  This little member    |  Argus implies no more, than     |
  |  gives life to every   |  the eye is in every part; that  |
  |  part about us; and I  |  is to say, every other part     |
  |  believe the story of  |  would be mutilated, were not    |
  |  Argus implies no      |  its force represented more by   |
  |  more, than the eye    |  the eye than even by itself.    |
  |  is in every part;     |  -- Addison                      |
  |  that is to say,       |                                  |
  |  every other part      |                                  |
  |  would be mutilated,   |                                  |
  |  were not its force    |                                  |
  |  represented more by   |                                  |
  |  the eye than even by  |                                  |
  |  itself. -- Addison    |                                  |
if ./gsl testabort ; then \
    echo "=== The gsl abort() returned a zero exit code, this is useless, FAIL"; \
    false ; \
else \
    echo "=== The gsl abort() returned non-zero exit code as expected, OK"; \
fi
GSL/4.1.5 Copyright (c) 1996-2016 iMatix Corporation
              GSL Developers 2016-2017
gsl/4 I: Processing testabort...
gsl/4 M: Error message goes here
=== The gsl abort() returned non-zero exit code as expected, OK
zhangna@loongson-pc:~/lucky-github/gsl/src$ arch 
loongarch64

```

**Review by zhuyaliang — APPROVED** [review]
编译无报错，运行正常

**Review by yetist — APPROVED** [review]
LGTM.

**Review by songding-sd — APPROVED** [review]
SGTM

**Comment by loongson-zn** [comment]
https://github.com/zeromq/gsl/pull/47

**Comment by xen0n** [comment]
Nice

**Pull request merged**